packages/meditrak-server/src/routes/userEntityPermissions/assertUserEntityPermissionPermissions.js

```diff
@@ -23,6 +27,52 @@ export const assertUserEntityPermissionPermissions = async (
   return true;
 };
 
+export const assertUserEntityPermissionEditPermissions = async (
+  accessPolicy,
+  models,
+  userEntityPermissionId,
+  updatedFields,
+) => {
+  // Check we have permission to access the record we're trying to edit
+  await assertUserEntityPermissionPermissions(accessPolicy, models, userEntityPermissionId);
+  // Check we have permission for the changes
+  await assertUserEntityPermissionUpsertPermissions(accessPolicy, models, updatedFields);
+
+  // Final check to make sure we're not editing a BES admin access permission
+  // There's nothing we could edit that couldn't be abused
+  const userEntityPermission = await models.userEntityPermission.findById(userEntityPermissionId);
+  const permissionGroup = await models.permissionGroup.findById(
```

Comment:
```suggestion
  const permissionGroup = await userEntityPermission.permissionGroup();
```
We often have this pattern when there's a foreign key relationship (have to manually set it up, so you need to check the model class each time, but in this case it definitely exists)